### PR TITLE
Solves # https://github.com/psi-probe/psi-probe/issues/1197

### PR DESCRIPTION
**Pull request by Aaron-LS**
Adds support for manually forcing a specific tomcat adapter via a properties file.

### CODE REVIEW

**Comment by hazendaz** [comment]
@Aaron-LS Can you please adhere to the 2-space formatting so this is easier to read.  Also, it appears that jdk11 is broken.  I'm not sure if that is a result of this PR or not but please check into that.  Otherwise, I'm good to bring something like this in to further security harden tomcat.  Thanks.

**Comment by Aaron-LS** [comment]
@hazendaz No problem, I have updated my fork to respect the 2 space standard. 

I looked at the jdk11 build failures and I am not sure they are related to my changes. They seem to be related to issues with a database connection:

`19:52:03.355 [C3P0PooledConnectionPoolManager[identityToken->1bqq1i09x8y3el7bpalnv|68c7ef83]-HelperThread-#1] DEBUG com.mchange.v2.resourcepool.BasicResourcePool - An exception occurred while acquiring a poolable resource. Will retry.
java.sql.SQLException: No suitable driver
	at java.sql/java.sql.DriverManager.getDriver(DriverManager.java:298)
	at com.mchange.v2.c3p0.DriverManagerDataSource.driver(DriverManagerDataSource.java:285)
	at com.mchange.v2.c3p0.DriverManagerDataSource.getConnection(DriverManagerDataSource.java:161)
	at com.mchange.v2.c3p0.WrapperConnectionPoolDataSource.getPooledConnection(WrapperConnectionPoolDataSource.java:161)
	at com.mchange.v2.c3p0.WrapperConnectionPoolDataSource.getPooledConnection(WrapperConnectionPoolDataSource.java:147)
	at com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool$1PooledConnectionResourcePoolManager.acquireResource(C3P0PooledConnectionPool.java:202)
	at com.mchange.v2.resourcepool.BasicResourcePool.doAcquire(BasicResourcePool.java:1138)
	at com.mchange.v2.resourcepool.BasicResourcePool.doAcquireAndDecrementPendingAcquiresWithinLockOnSuccess(BasicResourcePool.java:1125)
	at com.mchange.v2.resourcepool.BasicResourcePool.access$700(BasicResourcePool.java:44)
	at com.mchange.v2.resourcepool.BasicResourcePool$ScatteredAcquireTask.run(BasicResourcePool.java:1870)
	at com.mchange.v2.async.ThreadPoolAsynchronousRunner$PoolThread.run(ThreadPoolAsynchronousRunner.java:696)`

But I am pretty sure my changes shouldn't affect this.

**Comment by hazendaz** [comment]
Thanks I'll look further this weekend.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

File: core/src/main/java/psiprobe/beans/ContainerWrapperBean.java [hunk 1]
```diff
@@ -54,6 +68,9 @@
   @Inject
   private Map<String, ResourceResolver> resourceResolvers;
 
+  @Autowired
```
Comment (hazendaz) [inline core/src/main/java/psiprobe/beans/ContainerWrapperBean.java]: @Aaron-LS Please make this @Inject - spring prefers this when on the path for this type of injection.

File: core/src/main/java/psiprobe/beans/ContainerWrapperBean.java [hunk 2]
```diff
@@ -19,16 +19,27 @@
...
- * class, which in turn helps the Probe to interpret the wrapper.
+ * This class wires support for Tomcat "privileged" context functionality into
+ * Spring. If application context is privileged Tomcat would always call
+ * servlet.setWrapper method on each request. ContainerWrapperBean wires the
+ * passed wrapper to the relevant Tomcat container adapter class, which in turn
+ * helps the Probe to interpret the wrapper.
  */
+
+@Configuration
+@ComponentScan(basePackages = { "psiprobe" })
```
Comment (hazendaz) [inline core/src/main/java/psiprobe/beans/ContainerWrapperBean.java]: @Aaron-LS I don't think we need to list componentScan here.  It's already loaded in ProveConfig and set to scan the entire base package so I think this is just duplicate and possibly might affect resource processing.

**Comment by hazendaz** [comment]
I'm going to be out of town for a week so won't be able to come back to this right away.

**Comment by hazendaz** [comment]
@Aaron-LS Can you rebase this against recent master?

**Comment by hazendaz** [comment]
useful but closing as long out of date.

**Pull request closed without merging**